controller/users.go

```diff
@@ -77,6 +78,130 @@ func (c *UsersController) Show(ctx *app.ShowUsersContext) error {
 	})
 }
 
+func isServiceAccount(ctx context.Context) (bool, error) {
+	tokenManager, err := token.ReadManagerFromContext(ctx)
+	if err != nil {
+		return false, err
+	}
+	return (*tokenManager).IsServiceAccount(ctx), nil
+}
+
+// CreateUserAsServiceAccount creates a user when requested using a service account token
+func (c *UsersController) CreateUserAsServiceAccount(ctx *app.CreateUserAsServiceAccountUsersContext) error {
+
+	isSvcAccount, err := isServiceAccount(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": ctx.ID,
+		}, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to create a user.")))
+	}
+
+	return c.createUserInDB(ctx)
+}
+
+func (c *UsersController) createUserInDB(ctx *app.CreateUserAsServiceAccountUsersContext) error {
+
+	userID, err := uuid.FromString(ctx.Payload.Data.Attributes.UserID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("invalid user id")))
```

Comment:
Let's maybe finally add errors.NewBadRequest() function and start using it? Just for consistency.
But in this particular case I think we already have some error type which represents Bad Request in case invalid format...

Reply:
thanks.
now using `POST /users`  - therefore no BadRequest is being returned ever inside the controller.